### PR TITLE
Shrink `ast::Expr` from 72 bytes to 64 bytes

### PR DESCRIPTION
**Pull request by nnethercote**
r? @ghost

### CODE REVIEW

File: compiler/rustc_ast/src/ast.rs [hunk 1]
```diff
@@ -1450,11 +1464,13 @@ pub enum ExprKind {
...
     Underscore,
 
+    // njn: update this comment
     /// Variable reference, possibly containing `::` and/or type
     /// parameters (e.g., `foo::bar::<baz>`).
     ///
     /// Optionally "qualified" (e.g., `<Vec<T> as SomeTrait>::SomeType`).
-    Path(Option<P<QSelf>>, Path),
+    Path1(Path),
+    Path2(P<QSelf>, P<Path>),
```
Comment (ytmimi) [inline compiler/rustc_ast/src/ast.rs:1473]: Maybe `Path` and `QualifiedPath` are more descriptive names than `Path1` and `Path2`?

Comment (nnethercote) [reply compiler/rustc_ast/src/ast.rs:1473]: This is draft code, I will clean it up if the perf results indicate it's worth merging.

**Comment by nnethercote** [comment]
@Kobzol suggested that the cycle/wall-time changes are probably just noise(!), e.g. compare these recent CI perf runs (`check-full` only selected for easier comparison): [one](https://perf.rust-lang.org/compare.html?start=8b1dbf728add722d4db894b9b986ec24e1cdb0a1&end=eb909d81464900597b91f3bbf2a6bef10006c9ff&stat=cycles%3Au&showRawData=true&debug=false&opt=false&doc=false&incrUnchanged=false&incrFull=false&incrPatched=false), [two](https://perf.rust-lang.org/compare.html?start=0978711950b77582e4f8f334f6e9848d48ab7790&end=8b1dbf728add722d4db894b9b986ec24e1cdb0a1&stat=cycles%3Au&showRawData=true&incrFull=false&incrUnchanged=false&debug=false&opt=false&doc=false&incrPatched=false).

Yikes. So that takes away all the possible benefits. This doesn't seem worth pursuing, then.

**Pull request closed without merging**